lapack/gonum/dgecon.go

```diff
@@ -51,6 +53,9 @@ func (impl Implementation) Dgecon(norm lapack.MatrixNorm, n int, a []float64, ld
 	if anorm == 0 {
 		return 0
 	}
+	if math.IsNaN(anorm) {
+		return anorm
```

Comment:
I think it's worth having a comment here explaining that while the reference implementation treats the NaN anorm as invalid, we don't since it's not helpful to do so here.

Reply:
Done.